qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -223,22 +134,205 @@ def mcrz(
     """
     control_qubits = self.qbit_argument_conversion(q_controls)
     target_qubit = self.qbit_argument_conversion(q_target)
+
     if len(target_qubit) != 1:
         raise QiskitError("The mcrz gate needs a single qubit as target.")
     all_qubits = control_qubits + target_qubit
-    target_qubit = target_qubit[0]
     self._check_dups(all_qubits)
 
     n_c = len(control_qubits)
-    if n_c == 1:  # cu
-        _apply_cu(self, 0, 0, lam, control_qubits[0], target_qubit, use_basis_gates=use_basis_gates)
+    if n_c <= 6:
+        ncrz = ControlRotationGate(lam, n_c, axis="z")
```

Comment:
You can probably use 'QuantumCircuit.ucrz' with several id gates and one rz. Or maybe adapt ControlRotationGate to use ucrz. Nice observation that with a small number of qubits the multiplexer is a good approach.

Reply:
I think that mcrz should be in another PR, because the mcrz issue.